modules/sdp/SDP.js

```diff
@@ -136,6 +200,96 @@ export default class SDP {
         });
 
         this.raw = this.session + this.media.join('');
+
+        if (this.isP2P) {
+            return;
+        }
+
+        // For offers from Jicofo, a new m-line needs to be created for each new remote source that is added to the
+        // conference.
+        const newSession = transform.parse(this.raw);
+        const newMedia = [];
+
+        newSession.media.forEach(mLine => {
+            const type = mLine.type;
+
+            if (type === MediaType.APPLICATION) {
+                const newMline = cloneDeep(mLine);
+
+                newMline.mid = newMedia.length.toString();
+                newMedia.push(newMline);
+
+                return;
+            }
+
+            if (!mLine.ssrcs?.length) {
+                const newMline = cloneDeep(mLine);
+
+                newMline.mid = newMedia.length.toString();
+                newMedia.push(newMline);
+
+                return;
+            }
+
+            mLine.ssrcs.forEach((ssrc, idx) => {
+                // Do nothing if the m-line with the given SSRC already exists.
+                if (newMedia.some(mline => mline.ssrcs?.some(source => source.id === ssrc.id))) {
+                    return;
+                }
+                const newMline = cloneDeep(mLine);
+
+                newMline.ssrcs = [];
+                newMline.ssrcGroups = [];
+                newMline.mid = newMedia.length.toString();
+                newMline.bundleOnly = undefined;
+                newMline.direction = idx ? 'sendonly' : 'sendrecv';
+
+                // Add the sources and the related FID source group to the new m-line.
+                const group = mLine.ssrcGroups?.find(g => g.ssrcs.includes(ssrc.id.toString()));
```

Comment:
You are converting the ssrc id into a string many times here, calculate it once: `const idStr = ssrc.id.toString();` and then compare with it.